Common/Securities/Future/FuturesListings.cs

```diff
@@ -0,0 +1,225 @@
+        };
+
+        /// <summary>
+        /// Gets the listed futures contracts on a given date
+        /// </summary>
+        /// <param name="futureTicker">Ticker of the future contract</param>
+        /// <param name="time">Contracts to look up that are listed at that time</param>
+        /// <returns>The currently trading contracts on the exchange</returns>
+        public static Symbol[] ListedContracts(string futureTicker, DateTime time)
+        {
+            if (!_futuresListingRules.ContainsKey(futureTicker))
+            {
+                // No entries found. This differs from entries being returned as an empty array, where
+                // that would mean that no listings were found.
+                return null;
+            }
+
+            return _futuresListingRules[futureTicker](time);
+        }
+
+        /// <summary>
+        /// Gets contracts following a quarterly listing procedure, with a limit of
+        /// how many contracts are listed at once.
+        /// </summary>
+        /// <param name="canonicalFuture">Canonical Futures Symbol</param>
+        /// <param name="time">Contracts to look up that are listed at that time</param>
+        /// <param name="limit">Number of Symbols we get back/are listed at a given time</param>
+        /// <returns>Symbols that are listed at the given time</returns>
+        private static Symbol[] QuarterlyContracts(Symbol canonicalFuture, DateTime time, int limit)
+        {
+            var contractMonth = new DateTime(time.Year, time.Month, 1);
+            var futureExpiry = DateTime.MinValue;
+
+            // Skip any contracts that have already expired.
+            while (futureExpiry < time)
+            {
+                futureExpiry = FuturesExpiryFunctions.FuturesExpiryFunction(canonicalFuture)(contractMonth);
+                contractMonth = contractMonth.AddMonths(1);
+            }
+
+            // Negate the last incrementation from the while loop to get the actual contract month of the future.
+            var firstFutureContractMonth = contractMonth.AddMonths(-1);
+
+            var quarterlyContracts = new DateTime[limit];
+            // Gets the next closest month from the current month in multiples of 3
+            var quarterlyContractMonth = (int)Math.Ceiling((double)firstFutureContractMonth.Month / 3) * 3;
+
+            for (var i = 0; i < limit; i++)
+            {
+                quarterlyContracts[i] = firstFutureContractMonth.AddMonths(-firstFutureContractMonth.Month + quarterlyContractMonth);
+                quarterlyContractMonth += 3;
+            }
+
+            return quarterlyContracts
+                .OrderBy(t => t)
+                .Select(t => FuturesExpiryFunctions.FuturesExpiryFunction(canonicalFuture)(t))
+                .Where(e => e >= time)
+                .Select(e => Symbol.CreateFuture(canonicalFuture.ID.Symbol, canonicalFuture.ID.Market, e))
+                .ToArray();
```

Comment:
We could directly perform these operations in the loop above and avoid iterating twice, same for `MonthlyContractListings`, and we shouldn't need the order by?